src/arrayMethodReduce.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result;
+
+    if (initialValue || initialValue === 0) {
+      result = initialValue;
+
+      for (let i = 0; i < this.length; i++) {
+        result = callback(result, this[i], i, this);
+      }
+    }
+
+    if (!initialValue && initialValue !== 0) {
```

Comment:
```suggestion
    if (!initialValue) {
```

unnecessary another check. If there is NO initialValue, why do you also check if it's !==0?